### PR TITLE
Added support for abstract projections

### PR DESCRIPTION
**Pull request by PascalSenn**
Added support for Interface and Union types with projections

### CODE REVIEW

**Comment by AradAral** [comment]
Damn! You seriously managed to implement this in 3 hours! :D
I aspire to reach this productivity level as a programmer someday!
Great job!

File: src/HotChocolate/Data/src/Data/Projections/ISelectionVisitorContext.cs [hunk 1]
```diff
@@ -12,6 +12,8 @@ public interface ISelectionVisitorContext
 
         Stack<SelectionSetNode?> SelectionSetNodes { get; }
 
+        Stack<INamedType?> ResolvedType { get; }
```
Comment (PascalSenn) [inline src/HotChocolate/Data/src/Data/Projections/ISelectionVisitorContext.cs:15]: @michaelstaib  Should this wait till 12?

Comment (michaelstaib) [reply src/HotChocolate/Data/src/Data/Projections/ISelectionVisitorContext.cs:15]: would be better

Comment (kolpav) [reply src/HotChocolate/Data/src/Data/Projections/ISelectionVisitorContext.cs:15]: @michaelstaib Could this be merged and made part of preview release please?

**Comment by PascalSenn** [comment]
@michaelstaib  i cannot add the ready label from mobile

File: src/HotChocolate/Data/test/Data.Projections.SqlServer.Tests/ProjectionVisitorTestBase.cs [hunk 2]
```diff
@@ -49,23 +49,25 @@ public class ProjectionVisitorTestBase
...
-            Func<IResolverContext, IQueryable<TEntity>> resolver = BuildResolver(
-                onModelCreating,
-                entities);
+            Func<IResolverContext, IQueryable<TEntity>> resolver =
+                BuildResolver(onModelCreating, entities);
 
             ISchemaBuilder builder = SchemaBuilder.New();
 
-            if (objectType is {})
+            if (objectType is { })
```
Comment (michaelstaib) [inline src/HotChocolate/Data/test/Data.Projections.SqlServer.Tests/ProjectionVisitorTestBase.cs:64]: is not null?

Comment (michaelstaib) [inline src/HotChocolate/Data/test/Data.Projections.SqlServer.Tests/ProjectionVisitorTestBase.cs:64]:
```suggestion
            if (objectType is not null)
```

**Review by michaelstaib — APPROVED** [review]

**Pull request merged**